internal/kafka/internal/workers/clusters_mgr.go

```diff
@@ -447,14 +447,14 @@ func (c *ClusterManager) reconcileReadyCluster(cluster api.Cluster) error {
 		return errors.WithMessagef(err, "failed to reconcile ready cluster resources %s ", cluster.ClusterID)
 	}
 
-	err = c.reconcileClusterIdentityProvider(cluster)
```

Comment:
This reconciliation steps calls the `clusterService.GetClusterDNS()` which retrieves from the db or OCM is cluster dns is not set thus ended up performing the reconciliation step.
This shouldn't be needed to be done in this step and reconciling the cluster dns first by calling `reconcileClusterDNS(cluster)` ensures that the cluster dns will be persisted in the database.

Reply:
Looks good